### ISSUE TITLE
:Man on Void Linux ignores section

### ISSUE THREAD

**Issue by HiPhish** [neovim/neovim#17011]
### Neovim version (nvim -v)

0.6.1

### Vim (not Nvim) behaves the same?

n/a

### Operating system/version

Void Linux

### Terminal name/version

Alacritty 0.9.0

### $TERM environment variable

alacritty

### Installation

Void repository

### How to reproduce the issue

You need a man page which exists in multiple sections, such as `man(1)`, `man(1p)` (POSIX man) or `man(7)` (documentation of the man format)

- Execute `:Man` with explicit section, e.g. `:Man 7 man`

### Expected behavior

The man page with the corresponding section opens.

### Actual behavior

The man page with the first matching section (`1p` in my case) opens.

I noticed that the script calls `man -w -S 7 man`, which on my system prints

```
$ man -w -S 7 man
/usr/share/man/man1p/man.1p
/usr/share/man/man1/man.1
/usr/share/man/man7/man.7
```

I don't know about other distros, but on Void `-S` selects the machine architecture. The specify a section I have to use the `-s` option (lower-case s), which prints only one line: `/usr/share/man/man7/man.7`. Redefining the `s:section_arg` to `-s` is not enough though, so something else must be off as well.

**Comment by zeertzjq** [comment]
May be fixed by #13709

**Comment by zeertzjq** [comment]
Should be fixed now.
